src/Middleware/ThrottleMiddleware.php

```
@@ -195,7 +197,29 @@ protected function _touch(): int
             );
         }
 
-        return Cache::increment($this->_identifier, 1, static::$cacheConfig) ?: 0;
+        return Cache::increment($this->_identifier, $this->_getRequestWeight($request), static::$cacheConfig) ?: 0;
+    }
+
+    /**
+     * Returns configured weight of each request, result of callable, or 1 when option
+     * is misconfigured or the callable did not return integer >= 0
+     *
+     * @param \Psr\Http\Message\ServerRequestInterface $request ServerRequestInterface instance
+     * @return int
+     */
+    protected function _getRequestWeight(ServerRequestInterface $request): int
+    {
+        $configWeight = $this->getConfig('request_weight');
+
+        if (!is_int($configWeight) && !is_callable($configWeight)) {
+            return 1;
```

Never silently ignore mis-configuration, throw an exception here. Move the block below for valid callable above this.